CMakeLists.txt

```diff
@@ -24,6 +24,7 @@ include(CheckFunctionExists)
 # Platform shortcuts
 set_if(DARWIN      ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 set_if(LINUX       ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+set_if(BSD         ${CMAKE_SYSTEM_NAME} MATCHES "[bB][sS][dD]$")
```

Comment:
I know that CMake is poor as it comes to support case insensitive matching, but I really think this can be less clumsy:
```
string(TOLOWER ${CMAKE_SYSTEM_NAME} SYSNAME_LC)
set_if(BSD ${SYSNAME_LC} MATCHES "bsd$")
```
Similarly in line 702:
```
AND ${SYSNAME_LC} MATCHES "openbsd$"
```

Reply:
It seems clumsier to make a new variable. But I think that is just a measure of preference.

Reply:
I agree with @jlsantiago0 , single line statement, even with "[bB][sS][dD]$" in it, makes my mind twist less than tracing all that variable spaghetti.

Reply:
OK. I submitted a commit that uses a seperate variable for the lower case representation of the system name and use that for comparisons to support case variability.